Dockerfile

```diff
@@ -1,41 +1,84 @@
+ADD https://github.com/just-containers/s6-overlay/releases/download/v${S6_OVERLAY_VERSION}/s6-overlay-noarch.tar.xz /tmp
+
+# Add S6 optional symlinks
+ADD https://github.com/just-containers/s6-overlay/releases/download/v${S6_OVERLAY_VERSION}/s6-overlay-symlinks-noarch.tar.xz /tmp
+ADD https://github.com/just-containers/s6-overlay/releases/download/v${S6_OVERLAY_VERSION}/s6-overlay-symlinks-arch.tar.xz /tmp
+
+ENV LANG='en_US.UTF-8'                   \
+    LANGUAGE='en_US.UTF-8'               \
+    TERM='xterm'                         \
+    S6_LOGGING="1"                       \
+    S6_VERBOSITY="0"                     \
+    S6_CMD_WAIT_FOR_SERVICES_MAXTIME="0" \
+    TZ="Europe/London"
+
+RUN <<EOF
+    set -xe
+    apk upgrade --update --no-cache
+    tar -C / -Jxpf /tmp/s6-overlay.tar.xz
+    tar -C / -Jxpf /tmp/s6-overlay-noarch.tar.xz
+    tar -C / -Jxpf /tmp/s6-overlay-symlinks-noarch.tar.xz
+    tar -C / -Jxpf /tmp/s6-overlay-symlinks-arch.tar.xz
+
+    apk add --no-cache -U   \
+        bash                \
+        bash-completion     \
+        bash-doc            \
+        ca-certificates     \
+        curl                \
+        findmnt             \
+        fuse                \
+        libacl              \
+        libxxhash           \
+        logrotate           \
+        lz4-libs            \
+        mariadb-client      \
+        mariadb-connector-c \
+        mongodb-tools       \
+        openssl             \
+        pkgconfig           \
+        postgresql-client   \
+        sqlite              \
+        sshfs               \
+        tzdata
+    apk upgrade --no-cache
+EOF
+
+COPY --link requirements.txt /
+
+RUN --mount=type=cache,id=pip,target=/root/.cache,sharing=locked \
+    <<EOF
+    set -xe
+    python3 -m pip install -U pip
+    python3 -m pip install -Ur requirements.txt
+    borgmatic --bash-completion > "$(pkg-config --variable=completionsdir bash-completion)"/borgmatic
+EOF
+
+COPY --link root/ /
+
 VOLUME /root/.borgmatic
-VOLUME /etc/borgmatic.d
 VOLUME /root/.config/borg
-VOLUME /root/.ssh
 VOLUME /root/.cache/borg
-RUN apk add --update --no-cache \
-    bash \
-    bash-completion \
-    bash-doc \
-    ca-certificates \
-    curl \
-    findmnt \
-    fuse \
-    libacl \
-    logrotate \
-    lz4-libs \
-    mariadb-client \
-    mariadb-connector-c \
-    mongodb-tools \
-    openssl1.1-compat \
-    postgresql-client \
-    sqlite \
-    sshfs \
-    supercronic \
-    tzdata \
-    && rm -rf \
-    /var/cache/apk/* \
-    /.cache
-
-COPY --chmod=755 entry.sh /entry.sh
-COPY requirements.txt /
-
-RUN python3 -m pip install --no-cache -Ur requirements.txt
-RUN borgmatic --bash-completion > /usr/share/bash-completion/completions/borgmatic && echo "source /etc/bash/bash_completion.sh" > /root/.bashrc
-
-ENTRYPOINT ["/entry.sh"]
+
+HEALTHCHECK --interval=30s --timeout=10s --start-period=20s --retries=3 CMD borgmatic --version || exit 1
```

Comment:
Let me know if it would be helpful to have a dedicated healthcheck of some sort in borgmatic.

Reply:
> Let me know if it would be helpful to have a dedicated healthcheck of some sort in borgmatic.

If we can, that would be great! Not quite sure how one would do it, but if we can figure out a healthcheck which would check for base functionality of "app is at least up and running, and shouldn't have an issue with the current config", it would be fantastic!

Reply:
> I know I just offered to add a healthcheck, but maybe the existing `borgmatic config validate` already does what you're describing? If it doesn't in some way, I'm happy to change or add to it.

In your opinion, would that be a suitable check for the application (and would it output to the general log files)? The latter question is more a case of "we don't want to spam the logs if it's OK, and we want to make sure it exits the container with an appropriate exit code if it's not working as intended"

Reply:
You could always add `--verbosity -1` (errors only) if you didn't want any output from borgmatic other than errors. And then, yeah, it would probably be a suitable check. It of course won't tell you if any configured remote repositories are reachable or anything like that, but it's a good first step.

Reply:
Updated healthcheck in https://github.com/borgmatic-collective/docker-borgmatic/pull/303/commits/d3155314b384c7d14b17d63590e61dd705a20942

Have checked that exit code 1 is given when an invalid config is presented, and have also checked that the console does not get filled up even without the `verbosity -1` flag.

Reply:
Looks good! Just note that without `--verbosity -1`, borgmatic produces output when the configs files are valid. If that's not filling up the console, great.

Reply:
Aye, honestly thought it would fill up the console to be fair.

I did a curiosity test as with verbosity -1 it also doesn't give a healthcheck output. So it was kinda win/lose.

With no verbosity flag, it fills the healthcheck output, but not the console logs. So an unexpected win/win there.

Reply:
Ah gotcha, cool. Makes sense that healthcheck output would be separate.